src/test/java/com/codecool/ecosampler/service/SampleDataServiceTest.java

```diff
@@ -0,0 +1,92 @@
+package com.codecool.ecosampler.service;
+
+import com.codecool.ecosampler.controller.dto.sampledata.NewSampleData;
+import com.codecool.ecosampler.controller.dto.sampledata.SampleDataDTO;
+import com.codecool.ecosampler.domain.Form;
+import com.codecool.ecosampler.domain.Role;
+import com.codecool.ecosampler.domain.SampleData;
+import com.codecool.ecosampler.domain.User;
+import com.codecool.ecosampler.exception.NotFoundException;
+import com.codecool.ecosampler.repository.SampleDataRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class SampleDataServiceTest {
+    private UUID uuid;
+    private LocalDateTime localDateTime;
+    private User user;
+
+    @BeforeEach
+    public void setup() {
+        uuid = UUID.randomUUID();
+        localDateTime = LocalDateTime.now();
+        user = new User(uuid, "test", "test@test.cx", Role.SCIENTIST, "");
+    }
+
+    @Mock
+    private SampleDataRepository sampleDataRepository;
+    @Mock
+    private UserService userService;
+    @Mock
+    private FormService formService;
+    @Mock
+    private AnswerService answerService;
+    @InjectMocks
+    private SampleDataService sampleDataService;
+
+    //getSampleDataDTOByPublicId tests
+    @Test
+    void should_return_no_sampledata_dto() {
+        when(sampleDataRepository.findSampleDataByPublicId(any(UUID.class))).thenReturn(Optional.empty());
+        assertThrows(NotFoundException.class, () -> sampleDataService.getSampleDataDTOByPublicId(uuid));
+    }
+
+    @Test
+    void should_return_a_sampledata_dto() {
+        SampleData sampleData = new SampleData(uuid, localDateTime, user, new Form());
+        when(sampleDataRepository.findSampleDataByPublicId(any(UUID.class))).thenReturn(Optional.of(sampleData));
```

Comment:
```suggestion
        when(sampleDataRepository.findSampleDataByPublicId(uuid)).thenReturn(Optional.of(sampleData));
```
It should work like, if an exact id is passed then it returns the sampledata with that specific id.